### PR TITLE
[file-based CDK] Adding Excel format support

### PR DESCRIPTION
**Pull request by jhsb25**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-cdk/python/local-cdk-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+pandas==2.0.3
```
Comment (natikgadzhi) [inline airbyte-cdk/python/local-cdk-requirements.txt:1]: Curious, even if work in progress, why not just add them to the pyproject.toml temporarily?

File: airbyte-cdk/python/pyproject.toml [hunk 2]
```diff
@@ -34,6 +34,8 @@ isodate = "~0.6.1"
 Jinja2 = "~3.1.2"
 jsonref = "~0.2"
 jsonschema = "~3.2.0"
+pandas = "2.0.3"
```
Comment (natikgadzhi) [inline airbyte-cdk/python/pyproject.toml:37]: Oh, there they are.

We need to figure out if Pandas is okay to tug into the CDK. Could it be that all you need are specific pieces that it uses for xlsx, and they're in a package of their own?

File: airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py [hunk 3]
```diff
@@ -250,7 +250,11 @@ async def apply_python_development_overrides(context: ConnectorContext, connecto
         # Install the airbyte-cdk package from the local directory
         # We use --no-deps to avoid conflicts with the airbyte-cdk version required by the connector
         connector_container = connector_container.with_mounted_directory(f"/{path_to_cdk}", directory_to_mount).with_exec(
-            ["pip", "install", "--no-deps", f"/{path_to_cdk}"], skip_entrypoint=True
+            ["pip", "install", f"/{path_to_cdk}"], skip_entrypoint=True
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/actions/python/common.py:253]: If you have these deps in pyproject.toml, why is this part necessary?

**Review by natikgadzhi — COMMENTED** [review]
Great work @jhsb25, thank you for improving the CDK!

I'd love to work with you on this — can you explain some of the decisions (additional requirements file and pandas) and add some integration-style tests? I don't remember if we have fixtures in csv / different formats, but would be good to have an actual test file that the CDK would parse in a test, I think.

**Comment by marcosmarxm** [comment]
@jhsb25 did you have time to take a look in Natik's comments?

**Comment by jhsb25** [comment]
@natikgadzhi @marcosmarxm 
> @jhsb25 did you have time to take a look in Natik's comments?

working on them, will have fixes in this week

File: airbyte-cdk/python/pyproject.toml (hunk 2 above)
Comment (girarda) [reply airbyte-cdk/python/pyproject.toml:37]: I'm not strongly opposed, but we should add pandas to the [file-based extras](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/pyproject.toml#L79) since it's not required for most connectors

**Comment by marcosmarxm** [comment]
@jhsb25 do you still want to continue this contribution?
